utopia-remix/app/routes-test/v1.project.$id.spec.ts

```
@@ -119,3 +120,125 @@ describe('getProject', () => {
+      await createTestUser(prisma, { id: userId })
+      await createTestSession(prisma, { key: 'the-key', userId: userId })
+
+      projectProxy = jest.spyOn(proxyServer, 'proxy')
+      createProjectAccessMock = jest.spyOn(projectAccessModel, 'createProjectAccess')
+    })
+
+    it('should set access level for a new project', async () => {
+      projectProxy.mockResolvedValue({ id: projectId, ownerId: userId })
+      createProjectAccessMock.mockResolvedValue(null)
+      const req = newTestRequest({
+        method: 'PUT',
+        authCookie: 'the-key',
+        search: { accessLevel: 'public' },
+      })
+      const response = await (action({
+        request: req,
+        params: { id: projectId },
+        context: {},
+      }) as Promise<ApiResponse<{ id: string; projectId: string }>>)
+      const project = await response.json()
+      expect(project).toEqual({ id: projectId, ownerId: userId })
+      expect(createProjectAccessMock).toHaveBeenCalledWith({
+        projectId: projectId,
+        accessLevel: AccessLevel.PUBLIC,
+      })
+    })
+    it('should set access level to PRIVATE for a new project if not provided', async () => {
+      projectProxy.mockResolvedValue({ id: projectId, ownerId: userId })
+      createProjectAccessMock.mockResolvedValue(null)
+      const req = newTestRequest({
+        method: 'PUT',
+        authCookie: 'the-key',
+        search: {},
+      })
+      const response = await (action({
+        request: req,
+        params: { id: projectId },
+        context: {},
+      }) as Promise<ApiResponse<{ id: string; projectId: string }>>)
+      const project = await response.json()
+      expect(project).toEqual({ id: projectId, ownerId: userId })
+      expect(createProjectAccessMock).toHaveBeenCalledWith({
+        projectId: projectId,
+        accessLevel: AccessLevel.PRIVATE,
+      })
+    })
+    it('should set access level to PRIVATE for a new project if access level is invalid', async () => {
```

I think this is ok, but I'm wondering if we should hard fail the request immediately if the access level is invalid

---

Probably better that than creating things with the "wrong" access level.